stdlib/json/json.rbs

```diff
@@ -0,0 +1,360 @@
+# read and write. Plus, equally simple for machines to generate or parse. JSON
+# is completely language agnostic, making it the ideal interchange format.
+#
+# Built on two universally available structures:
+#     1. A collection of name/value pairs. Often referred to as an _object_, hash table, record, struct, keyed list, or associative array.
+#     2. An ordered list of values. More commonly called an _array_, vector, sequence or list.
+#
+# To read more about JSON visit: http://json.org
+#
+# ## Parsing JSON
+#
+# To parse a JSON string received by another application or generated within
+# your existing application:
+#
+#     require 'json'
+#
+#     my_hash = JSON.parse('{"hello": "goodbye"}')
+#     puts my_hash["hello"] => "goodbye"
+#
+# Notice the extra quotes `''` around the hash notation. Ruby expects the
+# argument to be a string and can't convert objects like a hash or array.
+#
+# Ruby converts your string into a hash
+#
+# ## Generating JSON
+#
+# Creating a JSON string for communication or serialization is just as simple.
+#
+#     require 'json'
+#
+#     my_hash = {:hello => "goodbye"}
+#     puts JSON.generate(my_hash) => "{\"hello\":\"goodbye\"}"
+#
+# Or an alternative way:
+#
+#     require 'json'
+#     puts {:hello => "goodbye"}.to_json => "{\"hello\":\"goodbye\"}"
+#
+# `JSON.generate` only allows objects or arrays to be converted to JSON syntax.
+# `to_json`, however, accepts many Ruby classes even though it acts only as a
+# method for serialization:
+#
+#     require 'json'
+#
+#     1.to_json => "1"
+#
+module JSON
+  # If *object* is string-like, parse the string and return the parsed result as a
+  # Ruby data structure. Otherwise generate a JSON text from the Ruby data
+  # structure object and return it.
+  #
+  # The *opts* argument is passed through to generate/parse respectively. See
+  # generate and parse for their documentation.
+  #
+  def self.[]: (untyped object, ?json_options opts) -> untyped
+
+  # This is create identifier, which is used to decide if the *json_create* hook
+  # of a class should be called. It defaults to 'json_class'.
+  #
+  def self.create_id: () -> _ToS
+
+  def self.create_id=: (_ToS create_id) -> _ToS
+
+  def self.deep_const_get: (_ToS path) -> untyped
+
+  def self.dump: (_ToJson obj, ?Integer limit) -> String
+               | (_ToJson obj, _JsonToWritableIO anIO) -> _JsonWrite
+               | (_ToJson obj, _JsonWrite anIO, ?Integer limit) -> _JsonWrite
+
+  # The global default options for the JSON.dump method:
+  #     :max_nesting: false
+  #     :allow_nan: true
+  #     :allow_blank: true
+  #
+  def self.dump_default_options: () -> json_options
+
+  def self.dump_default_options=: (json_options) -> json_options
+
+  def self.fast_generate: (_ToJson obj, ?json_options opts) -> String
+
+  alias self.fast_unparse self.fast_generate
+
+  def self.generate: (_ToJson obj, ?json_options opts) -> String
```

Comment:
Could you rewrite these module functions using `def self?.generate: ...` syntax?

Reply:
Thanks for the review! I updated the signature to use the `def self?.` syntax.